components/places/src/api/matcher.rs

```diff
@@ -68,14 +69,20 @@ pub fn match_url(conn: &PlacesDb, query: impl AsRef<str>) -> Result<Option<Strin
     }
 }
 
-fn match_with_limit(matchers: &[&dyn Matcher], max_results: u32) -> Result<(Vec<SearchResult>)> {
+fn match_with_limit(
+    conn: &PlacesDb,
```

Comment:
Can we hand this an `InterruptScope` instead of the entire connection?

Reply:
We could, although it makes sense to me that Connection would be an argument to search(), instead of having every implementation of Matcher borrow it though.